ambry-router/src/test/java/com.github.ambry.router/ChunkFillTest.java

```diff
@@ -319,20 +319,20 @@ private void fillChunksAndAssertSuccess() throws Exception {
   private void assertDataIdentity(ClusterMap clusterMap) throws IOException {
     if (!testEncryption) {
       ByteBuffer dest = ByteBuffer.allocate(totalSizeWritten);
-      for (ByteBuffer buf : compositeBuffers) {
+      for (ByteBuf buf : compositeBuffers) {
         Assert.assertNotNull("All chunks should have come in", buf);
-        buf.flip();
-        dest.put(buf);
+        for (ByteBuffer buffer: buf.nioBuffers()) {
```

Comment:
This is easier. dest will read all the ByteBufs in the array of compositeBuffers, so `buf.readBytes(dest)` would fail. We have to set limit on `dest` before reading bytes from `buf`. But in this way, we don't have to.